atom/browser/lib/desktop-capturer.coffee

```diff
@@ -0,0 +1,38 @@
+ipc = require 'ipc'
+BrowserWindow = require 'browser-window'
+
+# The browser module manages all desktop-capturer moduels in renderer process.
+desktopCapturer = process.atomBinding('desktop_capturer').desktopCapturer
+
+getWebContentsFromId = (id) ->
+  windows = BrowserWindow.getAllWindows()
+  return window.webContents for window in windows when window.webContents?.getId() == id
+
+# The set for tracking id of webContents.
+webContentsIds = new Set
+
+stopDesktopCapture = (id) ->
+  webContentsIds.delete id
+  # Stop updating if no renderer process listens the desktop capturer.
+  if webContentsIds.size is 0
+    desktopCapturer.stopUpdating()
+
+# Handle `desktopCapturer.startUpdating` API.
+ipc.on 'ATOM_BROWSER_DESKTOP_CAPTURER_START_UPDATING', (event, args) ->
+  id = event.sender.getId()
+  if not webContentsIds.has id
+    # Stop sending desktop capturer events to the destroyed webContents.
+    event.sender.on 'destroyed', ()->
+      stopDesktopCapture id
+  # Start updating the desktopCapturer if it doesn't.
+  if webContentsIds.size is 0
+    desktopCapturer.startUpdating args
```

Comment:
We can probably change the API to get all sources at once instead of getting update events:

```coffeescript
dc.getSources ['screen', 'window'], options, (sources) ->
  draw(source) for source in sources
```

So users can decide whether and how to refresh the thumbnails, and we don't need to provide options for the update period.

With this design the `getSources` call needs to start a new worker for each request, so Chrome's `DesktopMediaList` doesn't quite fit our use case, but it should be easy to reuse most of its code.